frontend/pkg/frontend/helpers_test.go

```diff
@@ -149,21 +152,25 @@ func TestCheckForProvisioningStateConflict(t *testing.T) {
 			name = fmt.Sprintf("%s (directState=%s)", tt.name, directState)
 			t.Run(name, func(t *testing.T) {
 				ctx := context.Background()
+				ctrl := gomock.NewController(t)
+				mockDBClient := mocks.NewMockDBClient(ctrl)
 
 				frontend := &Frontend{
-					dbClient: database.NewCache(),
+					dbClient: mockDBClient,
 				}
 
 				doc := database.NewResourceDocument(resourceID)
 				doc.ProvisioningState = directState
 
-				parentResourceID := resourceID.GetParent()
-				if parentResourceID.ResourceType.Namespace == resourceID.ResourceType.Namespace {
-					parentDoc := database.NewResourceDocument(parentResourceID)
-					// Hold the provisioning state to something benign.
-					parentDoc.ProvisioningState = arm.ProvisioningStateSucceeded
-					_ = frontend.dbClient.CreateResourceDoc(ctx, parentDoc)
-				}
+				mockDBClient.EXPECT().
+					GetResourceDoc(gomock.Any(), gomock.Any()).
+					DoAndReturn(func(ctx context.Context, resourceID *arm.ResourceID) (*database.ResourceDocument, error) {
+						resourceDoc := database.NewResourceDocument(resourceID)
+						// Hold the provisioning state to something benign.
+						resourceDoc.ProvisioningState = arm.ProvisioningStateSucceeded
+						return resourceDoc, nil
+					}).
```

Comment:
I think this is another case where we should use `Return` since we don't have any logic; we can define our `resourceDoc` in the test, then make a call to Return with it since it's the same every time.